frontend/src/components/TaskList.js

```diff
@@ -1,153 +1,120 @@
-import React, { Component } from 'react';
-import AddItem from './AddItem';
+import React, { useState, useEffect } from 'react';
+import AddTask from './AddTask';
 import Task from './Task';
 import { Typography } from '@material-ui/core';
 import Notification from './Notification';
-import Unauthorized from './Unauthorized';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import { fetchTasks, addTask, updateTask, deleteTask } from '@/store/actions/taskActions';
 
-class TaskList extends Component {
-    constructor(props) {
-        super(props);
+const TaskList = props => {
+    const [notifications, setNotifications] = useState([]);
+    const [isFetching, setIsFetching] = useState(true);
 
-        this.state = {
-            isLoggedIn: true,
-            notifications: [],
-            fetching: true
-        };
-    }
+    const { fetchTasks, tasks, addTask, updateTask, deleteTask } = props;
 
-    componentDidMount = async () => {
-        try {
-            await this.props.fetchTasks();
+    useEffect(() => {
+        fetchTasks();
 
-            this.setState({
-                fetching: false
-            });
-        } catch (error) {
-            console.error(error);
-            this.setState({ isLoggedIn: false });
-        }
-    };
+        setIsFetching(false);
```

Comment:
does the above action actually wait for tasks being fetched or is it immediately changing `isFetching`  to false? 

you should have `isFetching` or `isLoading` variable in reducer in store for tasks and get it here from store

Reply:
It was working, but refactored it